meta-mender-core/recipes-mender/mender/mender.inc

```diff
@@ -171,7 +182,8 @@ do_install() {
 
     #install configuration
     install -d ${D}/${sysconfdir}/mender
-    install -m 0644 ${B}/mender.conf ${D}/${sysconfdir}/mender
+    install -m 0644 ${B}/transient_mender.conf ${D}/${sysconfdir}/mender/mender.conf
+    install -m 0600 ${B}/data/persistent_mender.conf ${D}/${datadir}/mender/mender.conf
```

Comment:
The `/data` directory is our invention, and the `${datadir}` variable from Yocto is something completely different (`/usr/share` I believe). I don't think we have a dedicated variable for it, so just use `${D}/data` directly.

Also, you almost certainly will have to create the directory first, aka the `install -d` a couple of lines up.